ignition-server/controllers/tokensecret_controller.go

```diff
@@ -194,3 +196,42 @@ func decompress(content []byte) ([]byte, error) {
 	}
 	return data, nil
 }
+
+func getTokenIDTimeLived(tokenSecret *corev1.Secret) (*time.Duration, error) {
+	lastUpdated, ok := tokenSecret.Annotations[TokenSecretLastUpdatedTokenIDAnnotation]
+	if !ok {
+		return nil, nil
+	}
+
+	lasUpdatedTokenID, err := time.Parse(time.RFC3339, lastUpdated)
+	if err != nil {
+		return nil, err
+	}
+	timeLived := time.Since(lasUpdatedTokenID)
+	return &timeLived, nil
+
+}
+
+// tokenIDNeedRotation returns true if a duration is longer than the ttl/2.
+// This is the criteria to trigger a token ID rotation.
+func tokenIDNeedRotation(timeLived time.Duration) bool {
```

Comment:
should this take a pointer and return true if timeLived == nil ? (that would be the case for existing token secrets)